app/routines/find_or_create_task.rb

```diff
@@ -8,10 +8,19 @@ class FindOrCreateTask
   # otherwise saves and returns the task
   def exec(task)
 
-    outputs[:task] = Task.find_by(identifier: task.identifier,
-                                  resource: task.resource,
-                                  trial: task.trial) || task
-    outputs[:task].save unless outputs[:task].persisted?
+    lock_name = self.class.name
+
+    lock_result = Task.with_advisory_lock_result(lock_name, DEFAULT_LOCK_TIMEOUT_SECS) do
```

Comment:
This will lock the entire Task table whenever any Task is being created... I think maybe it is unnecessary (all `PG::TrWhatever` errors I've seen have been in the other routine, above).